src/app/mhc2gpdf.ml

```diff
@@ -1,28 +1,47 @@
 
-open Printf
+open Util
 
-let construct ofile ifile num_alt_to_add allele_lst notshort no_pdf no_open =
-  let ofile =
-    match ofile with
-    | None ->
-      begin
-        let base = Filename.basename ifile |> Filename.chop_extension in
-        match allele_lst with
-        | [] ->
-          begin
-            match num_alt_to_add with
-            | None   -> sprintf "%s_all" base
-            | Some n -> sprintf "%s_%d" base n
-          end
-        | lst        -> sprintf "%s_%d" base (List.length lst)
-      end
-    | Some s -> s
-  in
-  let short = not notshort in
-  let pdf   = not no_pdf in
-  let open_ = not no_open in
-  To_graph.construct ?num_alt_to_add allele_lst (Mas_parser.from_file ifile)
-  |> Ref_graph.output ~short ~pdf ~open_ ofile
+let construct ofile alignment_file num_alt_to_add allele_list notshort no_pdf no_open no_cache = function
```

Comment:
Additionally, it looks like the body code is almost exactly the same in each branch; might be easier to see what's different/what's going on in either branch by factoring out the same code?